packages/ra-core/src/store/useRemoveFromStore.ts

```diff
@@ -21,8 +21,18 @@ import { useStoreContext } from './useStoreContext';
 export const useRemoveFromStore = (hookTimeKey?: string) => {
     const { removeItem } = useStoreContext();
     return useCallback(
-        (key?: string) =>
-            removeItem(typeof key === 'undefined' ? hookTimeKey : key),
+        (key?: string) => {
+            if (
+                typeof key === 'undefined' &&
+                typeof hookTimeKey === 'undefined'
+            ) {
+                throw new Error(
+                    'You must provide a key to remove an item from the store'
+                );
+            }
+            // @ts-ignore
```

Comment:
TypeScript is dumb. It should know that either one of the two variables is defined, but its doesn't